src/scripts/main.js

```diff
@@ -1,3 +1,288 @@
+const buttonHowto = document.querySelector('.howto');
+const gameScore = document.querySelector('.game-score');
+const cells = document.querySelectorAll('td');
+
+const boardSideSize = 4;
+let score = 0;
+let isButtonStart = true;
+let isButtonHowto = true;
+const board = [];
+
+move();
+
+buttonHowto.addEventListener('click', e => {
+  if (isButtonHowto) {
+    e.target.innerHTML = 'Hide tips';
+    messageHowto.classList.remove('hidden');
+    isButtonHowto = false;
+  } else {
+    e.target.innerHTML = 'How to play';
+    messageHowto.classList.add('hidden');
+    isButtonHowto = true;
+  }
+});
+
+buttonStart.addEventListener('click', e => {
+  if (isButtonStart) {
+    e.target.className = 'button restart';
+    e.target.innerHTML = 'Restart';
+    messageStart.classList.add('hidden');
+    isButtonStart = false;
+
+    addNewTile();
+    addNewTile();
+  } else {
+    e.target.className = 'button start';
+    e.target.innerHTML = 'Start';
+    messageStart.classList.remove('hidden');
+    messageLose.classList.add('hidden');
+    messageWin.classList.add('hidden');
+    isButtonStart = true;
+    resetBoard();
+  }
+});
+
+function addNewTile() {
+  const randomNumber = (Math.random() < 0.1) ? 4 : 2;
+  const randomPlace = Math.floor(Math.random() * cells.length);
+
+  if (!cells[randomPlace].innerHTML) {
+    cells[randomPlace].innerHTML = randomNumber;
+    cells[randomPlace].className = `field-cell field-cell--${randomNumber}`;
+  } else {
+    addNewTile();
+    isGameOver();
+  }
+}
+
+function resetBoard() {
+  for (const cell of cells) {
+    for (let i = 0; i < 16; i++) {
+      cell.innerHTML = '';
+      cell.className = 'field-cell';
+    }
+  }
+}
+
+function updateBoard(tiles) {
+  gameScore.innerHTML = score;
+
+  tiles.forEach(item => {
+    (item.innerHTML)
+      ? item.className = `field-cell field-cell--${item.innerHTML}`
+      : item.className = 'field-cell';
+  });
+}
+
+function mergeInRow() {
+  let mergedTile;
+
+  for (let i = 0; i < cells.length - 1; i++) {
+    if (cells[i].innerHTML === cells[i + 1].innerHTML
+      && cells[i].parentElement === cells[i + 1].parentElement) {
+      mergedTile = parseInt(cells[i].innerHTML)
+      + parseInt(cells[i + 1].innerHTML);
+      cells[i].innerHTML = mergedTile;
+      cells[i + 1].innerHTML = '';
+
+      if (mergedTile === parseInt(mergedTile)) {
+        score += mergedTile;
+      };
+    }
+  }
+  isWinner();
+}
+
+function mergeInColumn() {
+  let mergedTile;
+
+  for (let i = 0; i < cells.length - 4; i++) {
+    if (cells[i].innerHTML === cells[i + boardSideSize].innerHTML) {
+      mergedTile = parseInt(cells[i].innerHTML)
+      + parseInt(cells[i + boardSideSize].innerHTML);
+      cells[i].innerHTML = mergedTile;
+      cells[i + boardSideSize].innerHTML = '';
+
+      if (mergedTile === parseInt(mergedTile)) {
+        score += mergedTile;
+      };
+    }
+  }
+  isWinner();
+}
+
+function isPossibleToMove() {
```

Comment:
```suggestion
function checkMovePossibility() {
```

Reply:
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

Reply:
and https://medium.com/swlh/how-to-better-name-your-functions-and-variables-e962a4ef335b